pkg/kube/inject/inject.go

```diff
@@ -473,6 +474,24 @@ func InjectionData(sidecarTemplate, valuesConfig, version string, typeMetadata *
 			return nil, "", merr
 		}
 	}
+
+	cluster := values.Map("global").Map("multiCluster").String("clusterName")
+	network := values.Map("global").String("network")
+	// for istiod-remote we use the injection URL to get the cluster and network names
+	if c, n := splitPath(path); c != "" && n != "" {
+		cluster, network = c, n
+	}
+	// explicit label takes highest precedence
+	if n, ok := metadata.Labels[label.IstioNetwork]; ok {
```

Comment:
Question: is the label set by users or by tools?

Reply:
Both. By default we inject, if a user wants to set it they can do that in the deployment.